Parsers/ASimWebSession/Parsers/ASIMWebSessionIIS.yaml

```diff
@@ -0,0 +1,110 @@
+  This ASIM parser supports normalizing IIS logs produced to the ASIM Web Session normalized schema.
+ParserName: ASIMWebSessionIIS
+EquivalentBuiltInParser: _ASIM_WebSession_IIS
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser = (disabled: bool = false)
+        {
+        W3CIISLog
+        | extend
+                EventResult = iff ( toint(scStatus) < 400, "Success", "Failure"),
+                scStatus = toint(scStatus), 
+                csUriQuery = iff(csUriQuery == "-", "", csUriQuery),
+                csUserName = iff(csUserName == "-", "", csUserName),
+                HttpVersion = iff((csVersion has "HTTP"), split(csVersion, "/")[1], "") // there is a limited chance that something connects over non-HTTP
+        | project-rename 
+                HttpHost = sSiteName,
+                HttpRequestMethod = csMethod,
+                HttpReferrer = csReferer,
+                User = csUserName, //probably won't have this one often
+                Dvc = Computer,
+                EventResultDetails = scStatus,
+                Dst = sIP,
+                Src = cIP,
+                UserAgent = csUserAgent,
+                ThreatCategory = IndicatorThreatType,
+                EventOriginalSeverity = Severity,
+                SrcGeoCountry = RemoteIPCountry,
+                SrcGeoLatitude = RemoteIPLatitude,
+                SrcGeoLongitude = RemoteIPLongitude,
+                ThreatOriginalConfidence = Confidence,
+                ThreatIpAddr = MaliciousIP,
+                ThreatIsActive = IsActive,
+                ThreatFirstReportedTime = FirstReportedDateTime,
+                ThreatLastReportedTime = LastReportedDateTime,
+                EventReportUrl = ReportReferenceLink,
+                EventOriginalSubType = scSubStatus
+        | extend
+                DstIpAddr = iff(csHost <> "", split(csHost, ":")[0], Dst),
+                DstNatIpAddr = iff(csHost <> "", Dst, ""),
+                EventType = 'WebServerSession',            
+                EventVendor = 'Microsoft',
+                EventSchemaVersion = '0.2.5',
+                EventSchema = 'WebSession', 
+                EventProduct = 'IIS',
+                DvcOs = 'Windows',
+                EventCount = int(1),
+                SrcIpAddr = Src,
+                IpAddr = Src,
+                HttpUserAgent = UserAgent,
+                HttpStatusCode = tostring(EventResultDetails),
+                EventStartTime = TimeGenerated,
+                EventEndTime = ( (TimeGenerated) + (TimeTaken * 1ms)), // TimeTaken field is in Milliseconds 
+                EventSeverity = iff(EventResult == "Success", "Informational", "Low"),
+                csUriQuery = tostring(csUriQuery),
+                csUriStem = tostring(csUriStem),
+                tempURLconstruct = iff(csUriQuery == "", csUriStem, strcat(csUriStem,"?",csUriQuery))
+        | extend                
+                Url = case( 
```

Comment:
The heuristic method of creating the URL is great! I think that breaking it down to steps would be both more efficient and more readable:

```
| extend 
    schema = iff (sPort contains "443", "https://", "http://"),
         // need to use contains nad not has
    port = iff (sPort in ("80", "443"), "", strcat (":", sPort))
    host = case ( 
        HttpHost == "Default Web Site", "",
        HttpHost != "", HttpHost,
        Dst
    )
 | extend 
    Url = strcat (schema, host, port, TempUrl)
```